File: competition_4/C.py
from collections import Counter

def main():
    import sys
    input = sys.stdin.readline

    testcases = int(input())

    for _ in range(testcases):
        n, k = (int(i) for i in input().strip().split())
        s = list(map(int, input().strip().split()))
        t = list(map(int, input().strip().split()))

        s_mods = Counter(e % k for e in s)

        possible = True
        for e in t:
            mod_class = e % k
            alt = (k - mod_class) % k

            if s_mods.get(mod_class, 0) > 0:
                s_mods[mod_class] -= 1
            elif s_mods.get(alt, 0) > 0:
                s_mods[alt] -= 1
            else:
                possible = False
                break
            
        if possible:
            print("YES")
        else:
            print("NO")
    
main()

File: competition_4/C.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int testcases;
    cin >> testcases;
    
    while (testcases--) {
        int n, k;
        cin >> n >> k;
        
        vector<int> s(n), t(n);
        for (int i = 0; i < n; i++) {
            cin >> s[i];
        }
        for (int i = 0; i < n; i++) {
            cin >> t[i];
        }
        
        unordered_map<int, int> s_mods;
        for (int e : s) {
            s_mods[e % k]++;
        }
        
        bool possible = true;
        for (int e : t) {
            int mod_class = e % k;
            int alt = (k - mod_class) % k;
            
            if (s_mods[mod_class] > 0) {
                s_mods[mod_class]--;
            } else if (s_mods[alt] > 0) {
                s_mods[alt]--;
            } else {
                possible = false;
                break;
            }
        }
        
        if (possible) {
            cout << "YES\n";
        } else {
            cout << "NO\n";
        }
    }
    
    return 0;
}